rclc/include/rclc/executor_handle.h

```diff
@@ -0,0 +1,194 @@
+  /// Total number of events
+  size_t number_of_events;
+} rclc_executor_handle_counters_t;
+
+
+/**
+ * Initializes the counters of each handle type to zero.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[inout] handle_counters preallocated rclc_executor_handle_counters_t
+ * \return `RCL_RET_INVALID_ARGUMENT` if `handle_counters` is a null pointer
+ */
+rcl_ret_t
+rclc_executor_handle_counters_zero_init(rclc_executor_handle_counters_t * handle_counters);
+
+/**
+ *  Initializes a handle with default values. The {@link rclc_executor_handle_t.index}
+ *  is initialized with `max_handles`, which is a non-valid index. Note that, valid indicies
+ *  are [0,max-handles-1]. The {@link rclc_executor_handle_t.invocation} is set to `ON_NEW_DATA`,
+ *  so that a potential callback is invoced only whenever new data is received. All other member
+ *  fields are set appropriate default values, like `none`, `NULL` or `false`.
+ *
+ *  * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[inout] handle preallocated rclc_executor_handle_t
+ * \param[in] max_handles maximum number of handles
+ * \return `RCL_RET_OK` if the \p handle was initialized successfully
+ * \return `RCL_RET_INVALID_ARGUMENT` if \p h is a null pointer
+ */
+rcl_ret_t
+rclc_executor_handle_init(
+  rclc_executor_handle_t * handle,
+  size_t max_handles);
+
+/**
+ *  Resets a rclc_executor_handle_t. Compared to the function  {@link rclc_executor_handle_init()}
+ *   only the {@link rclc_executor_handle_t.index} and {@link rclc_executor_handle_t.initialized}
+ *   variables are reset to default values.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[inout] handle a preallocated rclc_executor_handle_t
+ * \param[in] max_handles maximum number of handles
+ * \return `RCL_RET_OK` if \p h was cleared successfully
+ * \return `RCL_RET_INVALID_ARGUMENT` if \p h is a null pointer
+ */
+rcl_ret_t
+rclc_executor_handle_clear(
+  rclc_executor_handle_t * handle,
+  size_t max_handles);
+
+/**
+ *  Print out type name of a rclc_executor_handle_t.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[inout] handle preallocated rclc_executor_handle_t
+ * \return `RCL_RET_OK` if the handle was printed successfully
+ * \return `RCL_RET_INVALID_ARGUMENT` if \p h is a null pointer
+ */
+rcl_ret_t
+rclc_executor_handle_print(rclc_executor_handle_t * handle);
+
```

Comment:
```suggestion
```

Reply:
fixed in 54f0bbc54a33cc601764ab1d53fe3119d9e61c8b